os/arch/xtensa/src/esp32/esp32_i2c.c

```diff
@@ -646,21 +678,21 @@ static void i2c_esp32_get_i2c_data(struct esp32_i2c_priv_s *priv)
 0:  no slave address
 1: 7bit slave address
 2: 10bit slave address*/
-static void i2c_esp32_send_write_cmd(struct esp32_i2c_priv_s *priv, int addr_bytes, int cmd_start_index)
+static int i2c_esp32_send_write_cmd(struct esp32_i2c_priv_s *priv, int addr_bytes, int cmd_start_index)
 {
 	struct i2c_msg_s *msg = priv->transfer_msg;
-	uint32_t i = 0;
+	uint32_t i, to_send = 0;
 
 	volatile struct i2c_esp32_cmd *cmd = (void *)I2C_COMD0_REG(priv->i2c_num);
 	cmd += cmd_start_index;
 
 	if (msg->length == 0) {
 		*cmd = (struct i2c_esp32_cmd) {
 			.opcode = I2C_ESP32_OP_STOP,
-			.ack_en = false,
+			.ack_en = 0,
```

Comment:
Isn't it better using 'false' in readability perspective?

Reply:
ack_en is uint32_t, it's not a good choise to use bool value.